### PR TITLE
Add spec for OCaml opam packages

### PR DESCRIPTION
**Pull request by LaurentGoderre**

### CODE REVIEW

File: PURL-TYPES.rst [hunk 1]
```diff
@@ -498,6 +498,18 @@ including container images built by Docker and others:
       pkg:oci/static@sha256%3A244fd47e07d10?repository_url=gcr.io/distroless/static&tag=latest
       pkg:oci/hello-wasm@sha256%3A244fd47e07d10?tag=v1
 
+opam
+----
+``opam`` for OCaml packages
+- The default repository is ``https://opam.ocaml.org/``.
+- The ``name`` is the package name. It is not case sensitive and must be lowercased.
```
Comment (kit-ty-kate) [inline PURL-TYPES.rst:505]: package names are case sensitive and can use upper-case characters.
See the manual:
> Package names have the same restrictions as idents

https://opam.ocaml.org/doc/Manual.html#Package-Formulas

TL;DR:
```
<ident>         ::= { <identchar> }* <letter> { <identchar> }*
<identchar>     ::= <letter> | <digit>  | "_" | "-" | "+"
<letter>        ::= "a".."z" | "A".."Z"
<digit>         ::= "0".."9"
```

Comment (LaurentGoderre) [reply PURL-TYPES.rst]: @kit-ty-kate fixed!

**Review by kit-ty-kate — APPROVED** [review]

**Comment by pombredanne** [comment]
@LaurentGoderre here too.... after the merge of PR #514, PURL tests and defs are now defined in new JSON schemas :angel: :innocent: :grin: :
- See  #514

... therefore with the new approach... this PR would need to be updated. 


Do you think you can update this PR to the new format?

Sorry for the churn. :heart:

**Comment by pombredanne** [comment]
@LaurentGoderre or @kit-ty-kate do you mind to update the PR using the new JSON-based approach? That would be awesome! and quite timely with:
- https://ocaml.org/changelog/2025-10-03-security-team
- the CVE schema now having PURL supported merged https://github.com/CVEProject/cvelistV5/

Pinging the Ocaml security team lead too: @hannesm

**Comment by hannesm** [comment]
Thanks for the ping. Give me until the end of the week / next Monday to go over it. I also have a package-url implementtion in OCaml that I'm happy to share soon...
